packages/react-scripts/config/webpack.config.prod.js

```diff
@@ -162,7 +162,7 @@ module.exports = {
           },
           // Compile .tsx?
           {
-            test: /\.(ts|tsx)$/,
+            test: /\.(ts|tsx|js|jsx)$/,
```

Comment:
Hm... I don't think it's a good idea to use `ts-loader` for `jsx?`, while using `babel` for the same files when transpiling the tests. Since the dependencies used in upstream CRA for transpiling those files are already included for transpiling the tests, it should be favorable to use the `babel-loader` for this purpose. A simple copy/paste from upstream (maybe omitting the `thread-loader`? Dunno when this was added...) should be sufficient.

Reply:
Ah, no that's a very good point. I think we should load 'em all through ts-loader / ts-jest then. iirc ts-jest has no problems processing js. Thoughts?

Reply:
I'd still be concerned about potential differences from processing them with `babel` - CRA uses several plugins and preset configurations to transform the code (see [here](https://github.com/facebook/create-react-app/blob/next/packages/babel-preset-react-app/index.js#L48)), and not all of them are available using `typescript` (at least when not using `babel v7` with its `typescript` transformer). 
The synthesized default import that `babel` adds is just one of them, and projects being ported or just using `js(x)` files w.r.t. the official docs will definitely rely on this or any of the other transformations.

I'd favor to go the `babel` route for both `webpack` and `jest` due to this hardly predictable amount of potential inconsistencies.

Reply:
So if I'm understanding you right, what you're afraid of, is that the javascript code would behave different in CRA and CRA+TS ?
Because I'm reasonably sure that the typescript & JS files will behave reasonably identical between ts-jest and CRA. None of us are doing any crazy transforms on the code ( we have a babel step but that's only to hoist mock calls and stuff like that)

I don't think babel is the right way to go, we considered it as well (replacing ts-jest with just babel. See our discussion [here](https://github.com/kulshekhar/ts-jest/issues/358)

Reply:
> So if I'm understanding you right, what you're afraid of, is that the javascript code would behave different in CRA and CRA+TS ?

At least when users are migrating or re-using components that have been created to work with CRA or similar setups, yes.
Simple example - importing `react`:
```javascript
import React from "react";
```
The `react` module uses `module.exports`, thus there is no `default` field. As a result, this code won't compile using `typescript` without using `allowSyntheticDefaultImports`. However, since this flag does not affect the code itself, it will break at runtime, since the imported value will be `undefined`. `babel` synthesizes the `default` import.
To properly work with `typescript`, the correct way to import this would be:
```typescript
import * as React from "react";
```
But I doubt that those who were using `.js(x)` files before used imports like this. That's one of the most obvious issues that might occur. 
Additionally, you've already made `babel-jest` responsible for transpiling test files with `.jsx?` extension, so using `babel` resp. its loader is somewhat more reasonable for the build.

Don't get me wrong: I'd only take `babel` for `.js(x)` files, **not** for `.ts(x)`. Thus:
```
test: /\.tsx?$/ => ts-loader
test: /\.jsx?$/ => babel-loader
```
Additionally, this would be easier to maintain, since this rule can just be copied from upstream.

Reply:
Ah, I see. Some valid concerns. So babel-loader for jsx files, and babel-jest (iirc is the loader that's normally used) for jest?

Reply:
Yes, I suppose that's preferable.